### ISSUE TITLE
False positive ReservedWord in return type of phpdoc closure

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#1395]
Seen with the latest master branch

Expected: Does not emit ReservedWord because this is phpdoc
Observed: emits ReservedWord for the below code:

```php
<?php

/**
 * @param Closure():(resource|false) $op emits false positive ReservedWord
 * @return resource|false
 */
function create_resource($op) {
    return $op();
}
```
